svc.go

```diff
@@ -113,13 +144,15 @@ func (as *AutoNATService) handleDial(p peer.ID, obsaddr ma.Multiaddr, mpi *pb.Me
 		}
 	}
 
-	addrs := make([]ma.Multiaddr, 0)
+	addrs := make([]ma.Multiaddr, 0, AutoNATMaxPeerAddresses)
```

Comment:
_To improve my own Go knowledge_

What's the advantage of doing this VS:

```
var a []int
a = append(a, 1)
```

Reply:
If we we have an upper bound for how big the array will get that's sane, we can help go create a single allocations. repeated `append`s will repeatedly double the size of the array, and potentially need to copy the contents to a new location in memory as part of that reallocation.